x-pack/plugins/fleet/public/applications/fleet/sections/agent_policy/create_package_policy_page/index.tsx

```diff
@@ -323,7 +375,7 @@ export const CreatePackagePolicyPage: React.FunctionComponent = () => {
         history.push(packagePolicyPath);
       }
     },
-    [packagePolicy.policy_id, getPath, navigateToApp, history, routeState, wasNewAgentPolicyCreated]
+    [packagePolicy.policy_id, getPath, navigateToApp, history, routeState, queryParamsPolicyId]
   );
 
   const onSubmit = useCallback(async () => {
```

Comment:
It looks a little clunky to move it out to a different file, since it captures many dependencies from the component. I could refactor it to make it easier to read. Good point about the testing.